library/src-3.x/scala/TupledFunction.scala

```diff
@@ -0,0 +1,178 @@
+package scala
+
+import scala.annotation.implicitNotFound
+
+/** Type class relating a `FunctionN[..., R]` with an equvalent tupled function `Function1[TupleN[...], R]`
+ *
+ *  @tparam F a function type
+ *  @tparam G a tupled function type (function of arity 1 receiving a tuple as argument)
+ */
+@implicitNotFound("${F} cannot be tupled as ${G}")
+sealed trait TupledFunction[F, G] {
+  def tuple(f: F): G
+  def untuple(g: G): F
```

Comment:
What do you think about using past tenses? `tupled` and `untupled`.

Reply:
Good idea. Modified.